z/btree.go

```diff
@@ -65,24 +68,84 @@ func NewTree(tag string) *Tree {
 
 // NewTree returns a persistent on-disk B+ tree.
 func NewTreePersistent(path string) (*Tree, error) {
-	buffer, err := NewBufferPersistent(path)
+	t := &Tree{}
+	var err error
+
+	// Open the buffer from disk and set it to the maximum allocated size.
+	t.buffer, err = NewBufferPersistent(path, minSize)
 	if err != nil {
 		return nil, err
 	}
-	tree := &Tree{buffer: buffer}
-	tree.Reset()
-	return tree, nil
+	t.buffer.offset = uint64(len(t.buffer.buf))
+	t.data = t.buffer.Bytes()
+
+	// TODO(ajeet): can we do better than this?
+	isInitialized := false
+	for _, b := range t.data {
```

Comment:
You can just look at the root node, t.node(1), and check if that is initialized or not. If any part like pageID, numKeys, etc is non-zero for that node, then it is initialized.

Reply:
Since `pageID` can never be zero, we can just check for that, right? We shouldn't need to check anything else.

Reply:
Correct. This check is enough.